src/window.ts

```diff
@@ -262,8 +262,25 @@ export class Window {
             thisCell.yoff = thatCell.yoff
             thisCell.fontSize = thatCell.fontSize
             if (thisCell.t) {
-                thisCell.t.options.fontSize = thatCell.fontSize * this.gate.fontScale
-                thisCell.fit()
+                const hasFraction = String(thatCell.fontSize * this.gate.fontScale).includes('.')
+                thisCell.t.options.fontSize = Math.floor(thatCell.fontSize * this.gate.fontScale) + (hasFraction ? .5 : 0)
+                if (thisCell.fitAddon.proposeDimensions) {
+                    const dims = thisCell.fitAddon.proposeDimensions()
+                    if (dims && (dims.cols !== thatCell.cols || dims.rows !== thatCell.rows)) {
+                        thisCell.t.options.fontSize -= .5
+                    }
+                }
+                thisCell.t._core._renderService.clear()
+                thisCell.t.resize(thatCell.cols, thatCell.rows)
+                setTimeout(() => {
+                    if (thisCell instanceof Pane) {
```

Comment:
Please add a comment explaining this dark magic

Reply:
I'll rewrite this without the dark magic if I'm already using the _core anyways.